core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -253,14 +253,21 @@ public String toString() {
     for (HistoryEntry logEntry : snapshotLog) {
       if (last != null) {
         Preconditions.checkArgument(
-            (logEntry.timestampMillis() - last.timestampMillis()) >= 0,
+            (logEntry.timestampMillis() - last.timestampMillis()) >= -ONE_MINUTE,
             "[BUG] Expected sorted snapshot log entries.");
       }
       last = logEntry;
     }
+    if (last != null) {
```

Comment:
Rather than wrapping a precondition in an if, it's shorter to add an or:

```java
Preconditions.checkArgument(last == null || lastUpdatedMillis - last.timestampMillis() >= -ONE_MINUTE, ...);
```

Reply:
The issue is that the argument - `last.timestampMillis()` will be evaluated eagerly and may result in NPE

Reply:
It shouldn't be evaluated eagerly. Are you sure about that? We use [this pattern in other classes] (https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java#L131-L132) without a problem.